java/src/main/native/src/TableJni.cpp

```diff
@@ -656,6 +658,70 @@ std::vector<cudf::null_order> resolve_null_precedence(JNIEnv *env, jbooleanArray
 
 namespace {
 
+int set_column_metadata(cudf::io::column_in_metadata &column_metadata,
+                         std::vector<std::string> &col_names, 
+                         cudf::jni::native_jbooleanArray &nullability,
+                         cudf::jni::native_jbooleanArray &isInt96,
+                         cudf::jni::native_jintArray &precisions,
+                         cudf::jni::native_jintArray &children, int read_index) {
+  int write_index = 0;
+  int num_children = children[read_index++];
```

Comment:
My comment from https://github.com/rapidsai/cudf/pull/7998#discussion_r622431512 didn't get addressed.

Reply:
I got mixed up with the comment for passing the top_level_children as argument from Java. I have addressed it now.